File: app/javascript/channels/game_channel.js
import consumer from "channels/consumer"

window.app = {}
let params = new URLSearchParams(document.location.search);
let gameId = params.get("game_id"); // is the string "Jonathan"

window.app.action = consumer.subscriptions.create({ channel: "GameChannel", game_id: gameId }, {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("Hello");
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log(data)
    document.querySelector("#messages").innerHTML += `<p>${data.message}</p>`;
  }
});

File: app/channels/game_channel.rb
class GameChannel < ApplicationCable::Channel
  def subscribed
    game_id = params[:game_id]
    stream_from "game_channel_#{game_id}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def receive(data)
    # Example: Broadcast received data back to the channel (for testing purposes)
    game_id = params[:game_id]
    ActionCable.server.broadcast("game_channel_#{game_id}", data)
  end
end
